gbpservice/nfp/config_orchestrator/agent/loadbalancerv2.py

```diff
@@ -0,0 +1,457 @@
+
+from neutron_lbaas.db.loadbalancer import loadbalancer_dbv2
+
+from oslo_log import helpers as log_helpers
+from oslo_log import log as oslo_logging
+import oslo_messaging as messaging
+
+LOGGER = oslo_logging.getLogger(__name__)
+LOG = nfp_common.log
+
+"""
+RPC handler for Loadbalancer service
+"""
+
+
+class Lbv2Agent(loadbalancer_dbv2.LoadBalancerPluginDbv2):
+    RPC_API_VERSION = '1.0'
+    target = messaging.Target(version=RPC_API_VERSION)
+
+    def __init__(self, conf, sc):
+        super(Lbv2Agent, self).__init__()
+        self._conf = conf
+        self._sc = sc
+        self._db_inst = super(Lbv2Agent, self)
+
+    def _filter_service_info_with_resource(self, lb_db, core_db):
+        updated_db = {'subnets': [],
+                      'ports': []}
+        for lb in lb_db['loadbalancers']:
+            lb_port_id = lb['vip_port_id']
+            lb_subnet_id = lb['vip_subnet_id']
+            for subnet in core_db['subnets']:
+                if subnet['id'] == lb_subnet_id:
+                    updated_db['subnets'].append(subnet)
+            for port in core_db['ports']:
+                if port['id'] == lb_port_id:
+                    updated_db['ports'].append(port)
+        lb_db.update(updated_db)
+        return lb_db
+
+    def _to_api_dict(self, objs):
+        ret_list = []
+        for obj in objs:
+            ret_list.append(obj.to_api_dict())
+        return ret_list
+
+    def _get_core_context(self, context, tenant_id):
+        filters = {'tenant_id': [tenant_id]}
+        core_context_dict = common.get_core_context(context,
+                                                    filters,
+                                                    self._conf.host)
+        del core_context_dict['routers']
+        return core_context_dict
+
+    def _get_lb_context(self, context, filters):
+        args = {'context': context, 'filters': filters}
+        db_data = super(Lbv2Agent, self)
+        return {'loadbalancers': self._to_api_dict(
+                    db_data.get_loadbalancers(**args)),
+                'listeners': self._to_api_dict(
+                    db_data.get_listeners(**args)),
+                'pools': self._to_api_dict(
+                    db_data.get_pools(**args)),
+                'pool_members': self._to_api_dict(
+                    db_data.get_pool_members(**args)),
+                'healthmonitors': self._to_api_dict(
+                    db_data.get_healthmonitors(**args))}
+
+    def _context(self, **kwargs):
+        context = kwargs.get('context')
+        if context.is_admin:
+            kwargs['tenant_id'] = context.tenant_id
+        core_db = self._get_core_context(context, kwargs['tenant_id'])
+        # TODO(jiahao): _get_lb_context() fails for flavor_id, disable it
+        # for now. Sent the whole core_db to cofigurator
+        # lb_db = self._get_lb_context(**kwargs)
+        # db = self._filter_service_info_with_resource(lb_db, core_db)
```

Comment:
As memtioned in the known issues:
In config_orchestrator lbaasv2 agent, we are not able access neutron lb database due to this [issue](https://bugs.launchpad.net/neutron/+bug/1573337). Instead of filtering only relevant ports and subnets, I pass all ports and subnets under given tenant to configurator.

I would suggest leave it for now. We will revisit later